runtime/rasdump/javadump.cpp

```diff
@@ -5796,3 +5863,122 @@ getAllocatedVMThreadCount(J9JavaVM *vm)
 
 	return count;
 }
+
+/**
+ * This function is used to get the Redhat OS distribution details
+ * @param [out] OS Distribution Name
+ *
+ * @return returns 0 on success, any other -1 code on failure
+ */
+static int 
+getRedhatDistributionInfo(char *strDistributionVersion)
+{
+	int nSuccess = -1;
+	char strLine [OPENJ9_VERSION_INFO_SIZE] = "";
+	FILE *fpVersion = NULL;
+	/*read redhat Os distribution info file */
+	fpVersion = fopen(OPENJ9_REDHAT_OS_DISTRIBUTION_INFO_FILE, "r");
+	if (NULL != fpVersion) {
+		if(NULL != fgets(strLine, OPENJ9_VERSION_INFO_SIZE, fpVersion)) {
+			strcpy(strDistributionVersion, strLine);
+			fclose(fpVersion);
+			nSuccess = 0;
+		}
+	}
+	return nSuccess;
+}
+
+/**
+ * This function is used to get the SuSE OS distribution details
+ *@param [out] OS Distribution Name
+ *
+ * @return returns 0 on success, any other -1 code on failure
+ */
+static int 
+getSuseDistributionInfo(char *strDistributionVersion)
+{
+	int nSuccess = -1;
+	char strLine[OPENJ9_VERSION_INFO_SIZE] = "";
+	FILE *fpVersion = NULL;
+	/*read suse Os distribution info file */
+	fpVersion = fopen(OPENJ9_SUSE_OS_DISTRIBUTION_INFO_FILE, "r");
+	if (NULL != fpVersion) {
+		if(NULL != fgets(strLine, OPENJ9_VERSION_INFO_SIZE, fpVersion)) {
+			strcpy(strDistributionVersion, strLine);
+			fclose(fpVersion);
+			nSuccess = 0;
+		}
+	}
+    	return nSuccess;
+}
+/**
+ * This function is used to get the Ubuntu OS distribution details
+ * @param [out] OS Distribution Name
+ *
+ * @return returns 0 on success, any other -1 code on failure
+ */
+static int 
+getUbuntuDistributionInfo(char *strDistributionVersion)
+{
+	int nSuccess = -1;
+	char strLine[OPENJ9_VERSION_INFO_SIZE]="";
+	FILE *fpVersion=NULL;
+	fpVersion = fopen(OPENJ9_UBUNTU_OS_DISTRIBUTION_INFO_FILE, "r");
+	if (fpVersion!=NULL) {
+		while (NULL != fgets(strLine, OPENJ9_VERSION_INFO_SIZE, fpVersion)) {
+			if (NULL != strstr(strLine, OPENJ9_PRETTY_NAME) ) {
```

Comment:
When `strstr` returns a non-NULL value, it points at the desired string. There's no need to use `strtok` to find what follows, simply capture the return value and add the length of the prefix to find the value.

Please be consistent with formatting (the functions above are better). For example, there should be spaces either side of operators like `=` and `!=`; NULL should be on the left of comparisons; a space should appear after each comma.